x-pack/plugin/core/src/main/java/org/elasticsearch/license/RestGetTrialStatus.java

```diff
@@ -28,17 +22,8 @@
     }
 
     @Override
-    protected RestChannelConsumer doPrepareRequest(RestRequest request, XPackClient client) throws IOException {
-        return channel -> client.licensing().prepareGetStartTrial().execute(
-                new RestBuilderListener<GetTrialStatusResponse>(channel) {
-                    @Override
-                    public RestResponse buildResponse(GetTrialStatusResponse response, XContentBuilder builder) throws Exception {
-                        builder.startObject();
-                        builder.field("eligible_to_start_trial", response.isEligibleToStartTrial());
-                        builder.endObject();
-                        return new BytesRestResponse(RestStatus.OK, builder);
-                    }
-                });
+    protected RestChannelConsumer doPrepareRequest(RestRequest request, XPackClient client) {
+        return channel -> client.licensing().prepareGetStartTrial().execute(new RestToXContentListener<>(channel));
```

Comment:
ty for doing this on these responses. Does it need to be part of this PR any more now that we are not using the server libs for testing the client?

Reply:
it is not necessary as part of this PR, just a nice to have while I was at it.

Reply:
this is now necessary in order to use `AbstractHlrcStreamableXContentTestCase`